Added: Viem (Libraries) tool for developers (#625)
Consensys/doc.linea: docs/developers/tooling/libraries/viem.md

---
title: Viem
image: /img/socialCards/use-linea.jpg
---

> Viem is TypeScript Interface for Ethereum, Build reliable apps & libraries with lightweight, composable, and type-safe modules that interface with Ethereum.

Reviewer: ```suggestion
Viem is TypeScript Interface for Ethereum, enabling you to build reliable apps and libraries with lightweight, composable, and type-safe modules that interface with Ethereum.
```

> As the authors of wagmi, a popular React Hooks library for Ethereum, we struggled to work with the existing low-level TypeScript Ethereum libraries. We wanted to provide the users of wagmi with the best possible developer experience, <br/>

Reviewer: ```suggestion
As the authors of wagmi, a popular React Hooks library for Ethereum, we struggled to work with the existing low-level TypeScript Ethereum libraries. We wanted to provide the users of wagmi with the best possible developer experience.

```

> **Viem**: a TypeScript Interface for Ethereum that provides low-level stateless primitives for interacting with Ethereum. An alternative to ethers.js and web3.js with a focus on reliability, efficiency, and excellent developer experience.

Reviewer: this doesn't really follow on that well from the above statement. Maybe add a connecting statement like "Due to our experiences, we developed Viem [...]".

To point to your dapp on Linea, you need to use ```import { linea, lineaSepolia } from 'viem/chains'``` 'lineaSepolia', and `linea` for Linea Mainnet.

For more specifics, see here: [https://viem.sh/docs/chains/introduction](https://viem.sh/docs/chains/introduction). 

> See [Viem docs](https://viem.sh/docs/getting-started) for more info

Reviewer: ```suggestion
See the [Viem docs](https://viem.sh/docs/getting-started) for more information.
```
